src/dotnet/commands/dotnet-install/InstallCommand.cs

```diff
@@ -0,0 +1,137 @@
+
+            Reporter.Output.WriteLine(
+                $"{Environment.NewLine}The installation succeeded. If there is no other instruction. You can type the following command in shell directly to invoke: {commandName}");
+
+            return 0;
+        }
+
+        private static FilePath? GetConfigFileProbingCurrentDirectory()
+        {
+            // https://github.com/NuGet/Home/issues/6199
+            // Pending NuGet change
+
+            var settings = RestoreSettingsUtils.ReadSettings(
+                solutionDirectory: Directory
+                    .GetCurrentDirectory(), // current directory instead of fake project as start
+                restoreDirectory: Directory.GetCurrentDirectory(), // current directory instead of fake project as start
+                restoreConfigFile: null, // no explicit config file that why we prob it
+                machineWideSettings: new Lazy<IMachineWideSettings>(() => new XPlatMachineWideSetting()));
+
+            var nugetconfigfilePath = SettingsUtility.GetConfigFilePaths(settings).FirstOrDefault(); // it is ordered
+            if (nugetconfigfilePath != null)
+            {
+                return new FilePath(nugetconfigfilePath);
+            }
+
+            return null;
+        }
+
+        private static ToolConfigurationAndExecutableDirectory ObtainPackage(
+            string packageId,
+            string packageVersion,
+            FilePath? configFile,
+            string framework,
+            DirectoryPath executablePackagePath)
+        {
+            try
+            {
+                var toolPackageObtainer =
+                    new ToolPackageObtainer.ToolPackageObtainer(
+                        executablePackagePath,
+                        () => new DirectoryPath(Path.GetTempPath())
+                            .WithSubDirectories(Path.GetRandomFileName())
+                            .WithFile(Path.GetRandomFileName() + ".csproj"),
+                        new Lazy<string>(BundledTargetFramework.GetTargetFrameworkMoniker),
+                        new PackageToProjectFileAdder(),
+                        new ProjectRestorer());
+
+                return toolPackageObtainer.ObtainAndReturnExecutablePath(
+                    packageId: packageId,
+                    packageVersion: packageVersion,
+                    nugetconfig: configFile,
+                    targetframework: framework);
+            }
+            catch (PackageObtainException ex)
+            {
+                throw new GracefulException(
+                    message:
+                    $"Install failed. Failed to download package:{Environment.NewLine}{ex.Message}",
+                    innerException: ex);
+            }
+            catch (ToolConfigurationException ex)
+            {
+                throw new GracefulException(
+                    message:
+                    $"Install failed. Failed to apply the configuration of the tool:{Environment.NewLine}{ex.Message}",
```

Comment:
@KathleenDollard customer facing string. failed to restore package (package config is wrong)